plugin-dev/Source/Sentry/Sentry.Build.cs

```diff
@@ -1,16 +1,422 @@
+	{
+		string buildType = GetBuildType(target);
+
+		Console.WriteLine("Loading cmake target: "+target);
+
+		m_cmakeTargetPath=Path.GetFullPath(rules.Target.ProjectFile.FullName);
+		Console.WriteLine("Loading cmake path=: "+Directory .GetParent(m_cmakeTargetPath).FullName);
+		m_cmakeTargetPath=Directory .GetParent(m_cmakeTargetPath).FullName+"/Plugins/Sentry/sentry-native";
+
+		m_modulePath=Path.GetFullPath(rules.ModuleDirectory);
+		m_targetPath=Path.Combine(m_modulePath, m_targetLocation);
+
+		m_thirdPartyGeneratedPath=Path.Combine(rules.Target.ProjectFile.Directory.FullName, "Intermediate", "CMakeTarget");
+		m_generatedTargetPath=Path.Combine(m_thirdPartyGeneratedPath, m_targetName, m_targetPlatform);
+		m_buildDirectory = "build";
+		m_buildPath=Path.Combine(m_generatedTargetPath, m_buildDirectory);
+
+		m_buildInfoFile="buildinfo_"+buildType+".output";
+		m_buildInfoPath=Path.Combine(m_buildPath, m_buildInfoFile).Replace("\\", "/");
+    
+		if(!Directory.Exists(m_generatedTargetPath))
+			Directory.CreateDirectory(m_generatedTargetPath);
+
+		if(!Directory.Exists(m_buildPath))
+			Directory.CreateDirectory(m_buildPath);
+
+		var moduleBuilt = Build(target, rules, buildType);
+
+		if(!moduleBuilt)
+		{
+			return false;
+		}
+		return true;
+	}
+
+	private bool Build(ReadOnlyTargetRules target, ModuleRules rules, string buildType)
+	{
+		string builtFile = Path.Combine(m_generatedTargetPath, buildType+".built");
+		string projectCMakeLists=Path.GetFullPath(Path.Combine(m_targetPath, "CMakeLists.txt"));
+
+		bool configCMake=true;
+
+		//check if already built and CMakeList.txt not changed
+		if(File.Exists(builtFile))
+		{
+			DateTime cmakeLastWrite=File.GetLastWriteTime(projectCMakeLists);
+			string builtTimeString=System.IO.File.ReadAllText(builtFile);
+			DateTime builtTime=DateTime.Parse(builtTimeString);
+
+			if(builtTime.EqualsUpToSeconds(cmakeLastWrite))
+				configCMake=false;
+		}
+
+		if(configCMake)
+		{
+			Console.WriteLine("Target "+m_targetName+" CMakeLists.txt out of date, rebuilding");
+
+			var configureCommand = CreateCMakeConfigCommand(target, rules, m_buildPath, buildType);
+			var configureCode = ExecuteCommandSync(configureCommand);
+
+			if(configureCode!=0)
+			{
+				Console.WriteLine("Cannot configure CMake project. Exited with code: "
+				                  +configureCode);
+				return false;
+			}
+		}
+
+		var buildCommand = CreateCMakeBuildCommand(m_buildPath, buildType);
+		var buildCode = ExecuteCommandSync(buildCommand);
+
+		if(buildCode!=0)
+		{
+			Console.WriteLine("Cannot build project. Exited with code: "+buildCode);
+			return false;
+		}
+		else
+		{
+			if(configCMake)
+			{
+				DateTime cmakeLastWrite=File.GetLastWriteTime(projectCMakeLists);
+
+				File.WriteAllText(builtFile, cmakeLastWrite.ToString());
+			}
+		}
+		return true;
+	}
+	
+	private string GetWindowsGeneratorName(WindowsCompiler compiler)
+	{
+		string generatorName="";
+
+		switch(compiler)
```

Comment:
Do we cover all these branches in CI?

I don't think we're building on Windows yet, so we need to add that, for example.

Reply:
We do not currently build windows. i did talk to Ivan about this briefly when gathering information. Last info i got on the topic was during a weekly meeting it was mentioned he was pointed at some stuff that may help get it working.

As for the generators we currently dont build all the unreal generators even on linux. Im not even sure we try all cmake generators on the native side.  But the general goal here is to map the unreal compiler to its best fit in the cmake space.

Reply:
We might be better off adding all of the CI infrastructure first, before we continue with this change.
That way we can be confident we started off with something that works, and the changes continue to do so